ecl/hql/hqlgram2.cpp

```diff
@@ -12379,17 +12379,21 @@ IHqlExpression * PseudoPatternScope::lookupSymbol(IIdAtom * name, unsigned looku
 
 
 //---------------------------------------------------------------------------------------------------------------------
+#ifndef HQLEXPR_MULTI_THREADED
+static CriticalSection parseQueryCrit;
+#endif
+
 extern HQL_API IHqlExpression * parseQuery(IHqlScope *scope, IFileContents * contents, HqlLookupContext & ctx, IXmlScope *xmlScope, IProperties * macroParams, bool loadImplicit, bool isRoot)
 {
-    static CriticalSection parseQueryCrit; // Temporary - see HPCC-21986
-
     assertex(scope);
     try
     {
         if (isRoot)
             ctx.noteBeginQuery(scope, contents);
 
-        CriticalBlock b(parseQueryCrit); // Temporary - see HPCC-21986
+#ifndef HQLEXPR_MULTI_THREADED
```

Comment:
Should this be using HqlCriticalBlock ?

Reply:
no, because it is the opposite test - i.e. only protect parsing expressions if it is not enabled.